### PR TITLE
[libc] Move compile options to new cmake file

### PR DESCRIPTION
**Pull request by michaelrj-google**
The cmake test generator needed to be updated to support the same flags
as the source. To simplify the code, I moved it to a new file.

### CODE REVIEW

File: libc/cmake/modules/LLVMLibCCompileOptionRules.cmake
Comment (lntue) [inline libc/cmake/modules/LLVMLibCCompileOptionRules.cmake:1]: nit: extra empty line.

**Review by lntue — APPROVED** [review]

**Comment by jhuber6** [comment]
What is going on with all the commented out flags? This is going to break stuff if those aren't present. I would've preferred to hold off these rewrites until https://github.com/llvm/llvm-project/pull/81917 lands.

**Comment by michaelrj-google** [comment]
The commented out flags won't break anything, those are specifically in the newly added test compile options section. Those flags weren't being added at all before. The tests should still be passing (unless I've really screwed up)

I wanted to land this patch quickly since it should hopefully unblock @lntue 's own cmake rewrite. Hopefully it won't cause too many issues for your patch (which I'm in the middle of reviewing, I will hopefully be done soon), I tried to keep this change as small as reasonable.

**Pull request merged**
